tests/sles4sap/hana_cluster.pm

```diff
@@ -52,36 +72,43 @@ sub run {
         foreach ($node1, $node2) {
             add_to_known_hosts($_);
         }
-        assert_script_run "scp -qr /usr/sap/$sid/SYS/global/security/rsecssfs/* root\@$node2:/usr/sap/$sid/SYS/global/security/rsecssfs/";
-        assert_script_run qq(su - $sapadm -c "hdbsql -u system -p $sles4sap::instance_password -i $instance_id -d SYSTEMDB \\"BACKUP DATA FOR FULL SYSTEM USING FILE ('backup')\\""), 900;
-        assert_script_run "su - $sapadm -c 'hdbnsutil -sr_enable --name=$node1'";
+        assert_script_run "scp -qr /usr/sap/${sid}/SYS/global/security/rsecssfs/* root\@${node2}:/usr/sap/${sid}/SYS/global/security/rsecssfs/";
+        assert_script_run qq(su - ${sapadm} -c "hdbsql -u system -p $sles4sap::instance_password -i ${instance_id} -d SYSTEMDB \\"BACKUP DATA FOR FULL SYSTEM USING FILE ('backup')\\""), 900;
+        assert_script_run "su - ${sapadm} -c 'hdbnsutil -sr_enable --name=${node1}'";
 
         # Synchronize the nodes
-        barrier_wait "HANA_INIT_CONF_$cluster_name";
-        barrier_wait "HANA_CREATED_CONF_$cluster_name";
+        barrier_wait "HANA_INIT_CONF_${cluster_name}";
+        barrier_wait "HANA_CREATED_CONF_${cluster_name}";
+
+        hanasr_angi_hadr_providers_setup($sid, $instance_id, $sapadm) if get_var('USE_SAP_HANA_SR_ANGI');
 
-        # Upload the configuration into the cluster
-        my @crm_cmds = ("crm configure load update $cluster_conf",
-            "crm resource refresh msl_SAPHana_${sid}_HDB${instance_id}",
-            "crm resource maintenance msl_SAPHana_${sid}_HDB${instance_id} off");
+        # Commits configuration changes into the cluster
+        my $resource = get_var('USE_SAP_HANA_SR_ANGI') ? "mst_SAPHanaCtl_${sid}_HDB${instance_id}" : "msl_SAPHana_${sid}_HDB${instance_id}";
+        my @crm_cmds = ("crm configure load update ${cluster_conf}",
+            "crm resource refresh ${resource}",
+            'cs_wait_for_idle -s 5',
+            "crm resource maintenance ${resource} off");
         foreach my $cmd (@crm_cmds) {
             wait_for_idle_cluster;
             assert_script_run $cmd;
         }
     }
     else {
         # Synchronize the nodes
-        barrier_wait "HANA_INIT_CONF_$cluster_name";
+        barrier_wait "HANA_INIT_CONF_${cluster_name}";
 
-        assert_script_run "su - $sapadm -c 'sapcontrol -nr $instance_id -function StopSystem HDB'";
-        assert_script_run "until su - $sapadm -c 'hdbnsutil -sr_state' | grep -q 'online: false' ; do sleep 1 ; done", 120;
+        assert_script_run "su - ${sapadm} -c 'sapcontrol -nr ${instance_id} -function StopSystem HDB'";
+        assert_script_run "until su - ${sapadm} -c 'hdbnsutil -sr_state' | grep -q 'online: false' ; do sleep 1 ; done", 120;
```

Comment:
Same question about the added `{}` around variables.

Reply:
Curly braced variable is safely enclosed variable.